browse/services/listing/__init__.py

```diff
@@ -0,0 +1,125 @@
+"""Serves lists of articles for categories and time periods.
+
+Currently (2018-10) getting everything for a listing from the DB is
+not possible. There is no table that correctly records the publish
+history in the legacy DB.
+
+The legacy listing files are used only for the IDs of the papers
+announced. The rest of the metadata is not kept updated. An example of
+this causing a problem is if an article published on 2018-01-01, then
+crossed on 2018-01-02, then replaced with a differnt title on
+2018-01-03. The cross on 2018-01-02 in the listing file will have the
+old title.
+
+Why month granularity? The legacy listing files have only month
+granularity for when a paper was announced. In the future there might
+be better date granularity for new papers.
+"""
+
+
+from datetime import date
+from typing import List, Optional, Tuple
+
+from mypy_extensions import TypedDict
+
+
+ListingItem = TypedDict('ListingItem',
```

Comment:
This and the other ``TypeDict`` below would probably be good candidates for the domain.

Reply:
I agree

Reply:
These are not dataclasses, they are just type data for the dict that the listing service returns. 
I'll move them if you guys want them in some other file than where they are used, otherwise I'm going to leave them here.

Reply:
Take a look at https://arxiv.github.io/arxiv-arxitecture/crosscutting/services.html#data-domain for a description of the role of the domain. The short answer is that if they are crossing the API of the service integration module inside of the browse service, then they are probably part of the domain. Domain classes can be dataclasses, NamedTuples, type aliases, etc.

Reply:
Should this use just "from Flask import g" or "from arxiv.base.globals import get_application_global"?